### PR TITLE
Drop Python 3.6 support and address other smaller issues

### PR DESCRIPTION
**Pull request by netsettler**
* Drop support for Python 3.6

* Use `ignored`, `local_attrs`, and `override_environ`
  from `dcicutils.misc_utils` rather than `dcicutils.qa_utils`.

* Add debugging instrumentation for failed access to credentials.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -45,10 +44,10 @@ include = [
 
 [tool.poetry.dependencies]
 
-python = ">=3.6.1,<3.10"
+python = ">=3.7.0,<3.10"
 
 awscli = ">=1.18.174"
-dcicutils = ">=3.9.0"
+dcicutils = ">=3.9.0,<4"
```
Comment (drio18) [inline pyproject.toml:50]: Is capping the version here done out of precaution or are there incompatibilities that need to be addressed?

**Review by drio18 — CHANGES_REQUESTED** [review]
All changes here look good. However, I think there are 2 issues to address prior to merge:

- Existing references to using python 3.6 within the documentation that should be updated or dropped
- GA build is still using 3.6 so tests are failing

**Comment by netsettler** [comment]
> * Existing references to using python 3.6 within the documentation that should be updated or dropped
> * GA build is still using 3.6 so tests are failing

Good catches, thanks. I shoulda caught those. Not like it's the first time we've done this. Maybe I need a checklist, but I'd probably then just forget to check it.  Anyway, I'll get to it...

**Comment by netsettler** [comment]
@drio18, also, while we're being thorough, I should draw your attention to [SubmitCGAP needs to carefully upgrade to dcicutils 4.x (C4-896)](https://hms-dbmi.atlassian.net/browse/C4-896). I almost tried to do that as part of this PR, then backed out of it. Hopefully what I'm doing here will hold that issue constant and let us address it separately. Do let me know if you disagree.

**Pull request merged**